### PR TITLE
update main.js

### PR DESCRIPTION
**Pull request by throwarray**

### CODE REVIEW

File: data/shared/citizen/scripting/v8/main.js [hunk 1]
```diff
@@ -100,31 +101,55 @@ const EXT_LOCALFUNCREF = 11;
...
 		Citizen.invokeNative('0x91310870', name, dataSerialized, dataSerialized.length);
 	};
-	global.emitNet = (name, ...args) => {
-		const dataSerialized = pack(args);
 
-		Citizen.invokeNative('0x7fdd1128', name, dataSerialized, dataSerialized.length);
-	};
+	global.TriggerEvent = emit;
+
+	if (_in("0xcf24c52e", _r))  { // server
```
Comment (thers) [inline data/shared/citizen/scripting/v8/main.js]: It's ill advised to use these functions here, as they're part of automatic natives codegen (as in, can change w/o any notice).

File: data/shared/citizen/scripting/v8/main.js [hunk 2]
```diff
@@ -100,31 +101,55 @@ const EXT_LOCALFUNCREF = 11;
...
+		let TriggerClientEvent = function (eventName, playerId, ...args) {
```
Comment (thers) [inline data/shared/citizen/scripting/v8/main.js]: Why `let` and not `const`?

File: data/shared/citizen/scripting/v8/main.js [hunk 3]
```diff
@@ -100,31 +101,55 @@ const EXT_LOCALFUNCREF = 11;
...
 		emitter.on(name, callback);
 	};
-	global.on = global.addEventListener;
+
+	global.addEventListener = addEventListener;
+	global.on = addEventListener;
+
+	// global.RegisterNetEvent = function (name) { netSafeEventNames.add(name); };
+	// global.RegisterServerEvent = global.RegisterNetEvent;
+	global.AddEventHandler = addEventListener;
```
Comment (thers) [inline data/shared/citizen/scripting/v8/main.js]: Why exactly you need this alias?

File: data/shared/citizen/scripting/v8/main.js [hunk 4]
```diff
@@ -15,9 +15,10 @@ const EXT_LOCALFUNCREF = 11;
 
 	const pack = data => msgpack.encode(data, { codec });
 	const unpack = data => msgpack.decode(data, { codec });
-	
+
 	// store for use by natives.js
 	global.msgpack_unpack = unpack;
+	global.msgpack_pack = pack;
```
Comment (thers) [inline data/shared/citizen/scripting/v8/main.js]: Why you need this in global?

File: data/shared/citizen/scripting/v8/main.js [hunk 5]
```diff
@@ -112,8 +112,8 @@ const EXT_LOCALFUNCREF = 11;
 	global.addEventListener = addEventListener;
 	global.on = addEventListener;
 
-	global.RegisterNetEvent = function (name) { netSafeEventNames.add(name); };
-	global.RegisterServerEvent = global.RegisterNetEvent;
```
Comment (thers) [inline data/shared/citizen/scripting/v8/main.js]: And these are for backwards compatibility w/ emscripten-generated cpp resources.

Comment (throwarray) [reply data/shared/citizen/scripting/v8/main.js]: I added them in my first commit...

File: data/shared/citizen/scripting/v8/main.js (hunk 4 above)
Comment (throwarray) [reply data/shared/citizen/scripting/v8/main.js]: It's needed for packing function references among other things...

File: data/shared/citizen/scripting/v8/main.js (hunk 3 above)
Comment (throwarray) [reply data/shared/citizen/scripting/v8/main.js]: It was to match lua's API, I'll remove it I guess.

**Pull request closed without merging**